### ISSUE TITLE
 ERR_INVALID_ARG_TYPE

### ISSUE THREAD

**Issue by marl-on** [chegele/BDSAddonInstaller#6]
I run the command "bds-addon-installer" it successfully installed 3 out of 4 addons, but the last one gave a error.

Running Bedrock Dedicated Server Addon Installer...
[31/10/2020 20:02:27] GENERAL BDSAddonInstaller - Installing bAsilZadeaCanAlgbPack...
[31/10/2020 20:02:28] GENERAL BDSAddonInstaller - Successfully installed the bAsilZadeaCanAlgbPack pack.
[31/10/2020 20:02:28] GENERAL BDSAddonInstaller - Installing Multiplayer...
[31/10/2020 20:02:30] GENERAL BDSAddonInstaller - Successfully installed the Multiplayer pack.
[31/10/2020 20:02:30] GENERAL BDSAddonInstaller - Installing 7v40rOneplayerSleepSystemBP...
[31/10/2020 20:02:31] GENERAL BDSAddonInstaller - Successfully installed the 7v40rOneplayerSleepSystemBP pack.
[31/10/2020 20:02:31] GENERAL BDSAddonInstaller - Installing 7v41rOneplayerSleepSystemRP...
[31/10/2020 20:02:32] GENERAL BDSAddonInstaller - Successfully installed the 7v41rOneplayerSleepSystemRP pack.[31/10/2020 20:02:33] GENERAL BDSAddonInstaller - Installing lShowHPMOD...
internal/validators.js:92
        throw new ERR_INVALID_ARG_TYPE(name, 'number', value);
        ^

TypeError [ERR_INVALID_ARG_TYPE]: The "fd" argument must be of type number. Received undefined
    at Object.write (fs.js:641:3)
    at C:\Users\Marlon xdxd\AppData\Roaming\npm\node_modules\bds-addon-installer\node_modules\adm-zip\util\utils.js:160:44
    at FSReqCallback.oncomplete (fs.js:171:23) {
  code: 'ERR_INVALID_ARG_TYPE'
}
What could cause this error?

**Comment by chegele** [comment]
The adm-zip module, which is what threw the error, is used to extract the contents of the addons. The error suggest that it encountered some unexpected data while writing/saving one of the files in the addon. Would you be able to share a copy of the lShowHPMOD addon with me? 

I can provide some faster assistance in discord if you would prefer. 
http://discord.bedrockrealms.com

**Comment by TuyauEtanche** [comment]
Hey, I hate to bring this back up, but I figured out what happened when you get this error.

When you add a .mcaddon mod, sometimes the resulting .mcpack will be a little broken.
I opened the 2 resulting .mcpack, and found out that, in each pack, the program created a 1x loop to the same folder, but going from C:/ all the way to where the folder is, with each path being a folder itself.

**If anyone else gets this error**, simply open up the original .mcaddon file with 7-zip or WinRAR, and copy all the files from either the resource folder or the behavior folder, and paste them to the resulting .mcpack that you find in the BDS-Addon folder with the corresponding file (either the behavior or resource .mcpack, opened with 7-zip again)

This is probably a bug where the program gets confused with a specific filename or something, and it ends up adding the resulting position to the file name where it wants to copy the file, but it ends up adding a folder for each of the slashes ("/"), if that makes sense. So instead of copying the file to "C:/users/myUser/Desktop/BedrockServer/behavior_packs/Modpack.mcpack/items/customItem.json", it copies it like this: "C:/users/myUser/Desktop/BedrockServer/behavior_packs/Modpack.mcpack/items/C:/users/myUser/Desktop/BedrockServer/behavior_packs/Modpack.mcpack/items/customItem.json" (notice the duplicated path)

I hope that makes sense.
